Tasks/PublishTestResults/publishtestresults.ts

```diff
@@ -18,23 +19,42 @@ tl.debug('config: ' + config);
 tl.debug('testRunTitle: ' + testRunTitle);
 tl.debug('publishRunAttachments: ' + publishRunAttachments);
 
-//check for pattern in testResultsFiles
-if(testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
-  tl.debug('Pattern found in testResultsFiles parameter');
-  var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
-  var allFiles = tl.find(buildFolder);
-  var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
-}
-else {
-  tl.debug('No pattern found in testResultsFiles parameter');
-  var matchingTestResultsFiles = [testResultsFiles];
-}
+var matchingTestResultsFiles: string[] = getMatchingFiles(tl.getVariable('System.DefaultWorkingDirectory'), testResultsFiles, 'testResultsFiles');
 
-if(!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
-  tl.warning('No test result files matching ' + testResultsFiles + ' were found.');  
+if (!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
+  tl.warning('No test result files matching ' + testResultsFiles + ' were found.');
   tl.exit(0);
 }
-else{
+else {
   var tp = new tl.TestPublisher(testRunner);
   tp.publish(matchingTestResultsFiles, mergeResults, platform, config, testRunTitle, publishRunAttachments);
+}
+
+function getMatchingFiles(baseDirectory: string, inputPattern: string, inputPatternName: string): string[] {
```

Comment:
@vimegh there's already 'glob' module available in vsts-task-lib, which can find files from any path